src/ttsim/interface_dag_elements/specialized_environment_from_policy_inputs.py

```diff
@@ -0,0 +1,229 @@
+    labels__top_level_namespace: UnorderedQNames,
+    labels__grouping_levels: OrderedQNames,
+) -> SpecEnvWithoutTreeLogicAndWithDerivedFunctions:
+    """Return a flat policy environment with derived functions.
+
+    Two steps:
+    1. Remove all tree logic from the policy environment.
+    2. Add derived functions to the policy environment.
+
+    """
+    qname_env_without_tree_logic = _remove_tree_logic_from_policy_environment(
+        qname_env=dt.flatten_to_qnames(policy_environment),
+        labels__top_level_namespace=labels__top_level_namespace,
+    )
+    return _add_derived_functions(
+        qname_env_without_tree_logic=qname_env_without_tree_logic,
+        tt_targets=(
+            (labels__all_qnames_in_policy_environment | set(tt_targets__qname))
+            - set(labels__input_columns)
+        ),
+        input_columns=qnames_to_derive_functions_from,
+        grouping_levels=labels__grouping_levels,
+    )
+
+
+@interface_function()
+def without_processed_data_nodes_with_dummy_callables(
+    without_tree_logic_and_with_derived_functions: SpecEnvWithoutTreeLogicAndWithDerivedFunctions,  # noqa: E501
+    labels__input_columns: UnorderedQNames,
+) -> SpecEnvWithoutTreeLogicAndWithDerivedFunctions:
+    """Remove nodes that are in the processed data from the policy environment."""
+    return {
+        qn: dummy_callable(obj=n, leaf_name=dt.tree_path_from_qname(qn)[-1])
+        if not callable(n)
+        else n
+        for qn, n in without_tree_logic_and_with_derived_functions.items()
+        if qn not in labels__input_columns
+    }
+
+
+@interface_function()
+def complete_tt_dag(
+    without_processed_data_nodes_with_dummy_callables: SpecEnvWithoutTreeLogicAndWithDerivedFunctions,  # noqa: E501
+    tt_targets__qname: OrderedQNames,
+    labels__all_qnames_in_policy_environment: UnorderedQNames,
+    labels__input_columns: UnorderedQNames,
+) -> nx.DiGraph:
+    """Create the complete DAG.
+
+    The DAG is based on without_tree_logic_and_with_derived_functions because it should
+    include parameters and param_functions. Transforms non-callable nodes into callables
+    to include them as nodes in the DAG.
+    """
+    return create_dag(
+        functions=without_processed_data_nodes_with_dummy_callables,
+        targets=(
+            (labels__all_qnames_in_policy_environment | set(tt_targets__qname))
+            - set(labels__input_columns)
+        ),
+    )
+
+
+@interface_function()
+def with_processed_params_and_scalars(
+    without_tree_logic_and_with_derived_functions: SpecEnvWithoutTreeLogicAndWithDerivedFunctions,  # noqa: E501
+    backend: Literal["numpy", "jax"],
+    xnp: ModuleType,
+    dnp: ModuleType,
+    evaluation_date: datetime.date | None,
+) -> SpecEnvWithProcessedParamsAndScalars:
+    return specialized_environment.with_processed_params_and_scalars(
+        without_tree_logic_and_with_derived_functions=without_tree_logic_and_with_derived_functions,
+        processed_data={},
```

Comment:
I am not sure this should work because in `specialized_environment.with_processed_params_and_scalars`, we are removing the contents of `processed_data` from the graph. This will leave all nodes in there, which is probably behind the failure. Hence my comment on creating some fake processed data.